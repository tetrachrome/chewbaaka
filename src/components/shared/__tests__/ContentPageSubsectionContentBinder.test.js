/**
 * ContentPageSubsectionContentBinder.test.js
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Jul 15, 2020
 * Updated  : Jul 15, 2020
 */

import React from 'react'

import renderer from 'react-test-renderer';

import {
  ContentPageSubsectionParagraphsContentBinder,
  ContentPageSubsectionColumnDataBinderWithParagraphsContentBinder
} from '../ContentPageSubsectionContentBinder'

const paragraphContents = {
  paragraph_01: "Hello, World!",
  paragraph_02: "Cheetahs are such amazing animals!",
  paragraph_03: "We need to do our best to save them."
}

const columnData = {
  title: "Title",
  subtitle: "This is a subtitle",
  content: paragraphContents
}

test('ContentPageSubsectionParagraphsContentBinder snapshot', () => {
  const tree = renderer
    .create(
      <div>
        {ContentPageSubsectionParagraphsContentBinder(paragraphContents)}
      </div>
    ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('ContentPageSubsectionColumnDataBinderWithParagraphsContentBinder snapshot', () => {
  const renderedColumnData = ContentPageSubsectionColumnDataBinderWithParagraphsContentBinder(columnData);

  expect(renderedColumnData.title).toBe(columnData.title);
  expect(renderedColumnData.subtitle).toBe(columnData.subtitle);

  const tree = renderer
    .create(
      <div>
        {renderedColumnData.content}
      </div>
    ).toJSON();
  expect(tree).toMatchSnapshot();
});
