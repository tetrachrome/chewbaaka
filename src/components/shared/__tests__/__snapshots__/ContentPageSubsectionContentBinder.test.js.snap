// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentPageSubsectionColumnDataBinderWithParagraphsContentBinder snapshot 1`] = `
<div>
  <p
    className="ContentPageRegularTextSize ContentPageRegularTextSize_DebugColor DebugWireframe ContentPageRegularTextLineHeight ContentPageRegularTextLineHeight_DebugColor DebugWireframe ContentPageContentParagraphText ContentPageContentParagraphText_DebugColor DebugWireframe"
  >
    Hello, World!
  </p>
  <p
    className="ContentPageRegularTextSize ContentPageRegularTextSize_DebugColor DebugWireframe ContentPageRegularTextLineHeight ContentPageRegularTextLineHeight_DebugColor DebugWireframe ContentPageContentParagraphText ContentPageContentParagraphText_DebugColor DebugWireframe"
  >
    Cheetahs are such amazing animals!
  </p>
  <p
    className="ContentPageRegularTextSize ContentPageRegularTextSize_DebugColor DebugWireframe ContentPageRegularTextLineHeight ContentPageRegularTextLineHeight_DebugColor DebugWireframe ContentPageContentParagraphText ContentPageContentParagraphText_DebugColor DebugWireframe"
  >
    We need to do our best to save them.
  </p>
</div>
`;

exports[`ContentPageSubsectionParagraphsContentBinder snapshot 1`] = `
<div>
  <p
    className="ContentPageRegularTextSize ContentPageRegularTextSize_DebugColor DebugWireframe ContentPageRegularTextLineHeight ContentPageRegularTextLineHeight_DebugColor DebugWireframe ContentPageContentParagraphText ContentPageContentParagraphText_DebugColor DebugWireframe"
  >
    Hello, World!
  </p>
  <p
    className="ContentPageRegularTextSize ContentPageRegularTextSize_DebugColor DebugWireframe ContentPageRegularTextLineHeight ContentPageRegularTextLineHeight_DebugColor DebugWireframe ContentPageContentParagraphText ContentPageContentParagraphText_DebugColor DebugWireframe"
  >
    Cheetahs are such amazing animals!
  </p>
  <p
    className="ContentPageRegularTextSize ContentPageRegularTextSize_DebugColor DebugWireframe ContentPageRegularTextLineHeight ContentPageRegularTextLineHeight_DebugColor DebugWireframe ContentPageContentParagraphText ContentPageContentParagraphText_DebugColor DebugWireframe"
  >
    We need to do our best to save them.
  </p>
</div>
`;
